### PR TITLE
[NPU] [DLLM] Support openPangu_Diffusion DLLM model

### PR DESCRIPTION
**Pull request by lixinqi7**
## Motivation

This PR introduces support for openPangu_Diffusion, a diffusion-based language model architecture, to SGLang's DLLM algorithm suite.

While sharing some conceptual similarities with Llada2.0, openPangu_Diffusion exhibits distinct architectural requirements regarding logit alignment and attention masking. This PR provides the necessary specialized implementation to ensure efficient and accurate inference for this model family.

These are the links related to the model：
https://ai.gitcode.com/ascend-tribe/openPangu-7B-Diffusion-Base?source_module=search_result_model
https://ai.gitcode.com/ascend-tribe/openPangu-R-7B-Diffusion?source_module=search_result_model

## Modifications

1. Logits Right Shift Logic
Unlike Llada2.0, the output logits of openPangu_Diffusion require an explicit right-shift operation. 

2. Causal Attention during KV Cache Saving
In the final stage of saving the KV Cache, we have enforced a Causal Attention Mask. 

3. Left Padding Strategy
To strictly decouple the Prefill and Decode phases, we implemented a Left Padding strategy. 

## How to run
1、To use this feature, create a configuration file and load it using the --dllm-algorithm-config flag.
Example: dllm_config.yaml
alg: "confidence_threshold"  # entropy or confidence_threshold 
threshold: 0.9          # Confidence threshold for token filtering
block_size: 32
num_small_blocks: 4     
pad_id: 0               # Model specific padding ID
eos_token_id: 2     # Set to 2 for Base model or 45892 for R model

2、Launch Command
python -m sglang.launch_server \
  --model-path /openPangu-7B-Diffusion-Base \
  --trust-remote-code \
  --attention-backend ascend \
  --device npu \
  --mem-fraction-static 0.8 \
  --host 0.0.0.0 \
  --port 30000 \
  --dllm-algorithm "ShiftConfidence" \
  --dllm-algorithm-config dllm_config.yaml \
  --skip-server-warmup \
  --max-running-requests 1 \
  --enable-tokenizer-batch-encode \
  --dtype bfloat16 \
  --kv-cache-dtype bfloat16 \
  --disable-radix-cache \
  --disable-cuda-graph

## Accuracy Tests
The test results are available for reference at the provided GitCode link.

Note: If you use /sglang/benchmark to test the accuracy of the openpangu-R-7B-Diffusion model, you need to add a chat template to the test script.

## Known Limitations
To ensure stability in the current implementation, please note the following constraints:

CUDA Graph / Graph Mode: Currently not supported. Please ensure --disable-cuda-graph is used (especially on Ascend NPU).

Batch Size: Currently supports Batch Size = 1 only (--max-running-requests 1). Requests with batch sizes greater than 1 may lead to unexpected behavior in the ShiftConfidence logic.

## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [ ] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review Process

1. Ping Merge Oncalls to start the PR flow. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - `/tag-run-ci-label`, `/rerun-failed-ci`, `/tag-and-rerun-ci`
4. After green CI and required approvals, ask Merge Oncalls to merge.

### CODE REVIEW

**Comment by hnyls2002** [comment]
Thanks @lixinqi7! This has been conflicting since February and the code it patches has changed shape - `python/sglang/srt/dllm/config.py:36` replaced the per-architecture if/elif chain with a `DLLM_PARAMS` dict, and `python/sglang/srt/layers/rotary_embedding.py` is now a package rather than a file. openPangu support is still absent from main, so a rebased PR would be welcome. Closing this one - please reopen if I've missed something.
